openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/TestCaseRepository.java

```diff
@@ -660,9 +660,10 @@ public RestUtil.DeleteResponse<TestCase> deleteTestCaseFromLogicalTestSuite(
     deleteRelationship(testSuiteId, TEST_SUITE, testCaseId, TEST_CASE, Relationship.CONTAINS);
     // remove test case from logical test suite summary and update test suite
     removeTestCaseFromTestSuiteResultSummary(testSuiteId, testCase.getFullyQualifiedName());
-    EntityReference entityReference =
-        Entity.getEntityReferenceById(TEST_SUITE, testSuiteId, Include.ALL);
-    testCase.setTestSuite(entityReference);
+    TestCase updatedTestCase = Entity.getEntity(Entity.TEST_CASE, testCaseId, "*", Include.ALL);
```

Comment:
do we need to all of these on the delete path. Wouldn't this run into an issue if there is permanent deletion?

Reply:
Here we are not deleting the test case, just removing it from a logical test suite.

Reply:
LMK if that makes sense